engine/classes/Elgg/I18n/Translator.php

```diff
@@ -61,22 +61,31 @@ public function translate($message_key, $args = [], $language = "") {
 
 		$this->ensureTranslationsLoaded($language);
 
-		if (isset($GLOBALS['_ELGG']->translations[$language][$message_key])) {
-			$string = $GLOBALS['_ELGG']->translations[$language][$message_key];
+		$notice = '';
+		$string = $message_key;
 
-		} else if (isset($GLOBALS['_ELGG']->translations["en"][$message_key])) {
-			$string = $GLOBALS['_ELGG']->translations["en"][$message_key];
-			_elgg_services()->logger->notice(sprintf('Missing %s translation for "%s" language key', $language, $message_key));
+		foreach ([$language, 'en'] as $try_lang) {
```

Comment:
this will loop twice if `$language` already is 'en' .. shouldn't it be a unique array?